Branch-SDK/src/io/branch/referral/RemoteInterface.java

```diff
@@ -119,10 +119,12 @@ public ServerResponse make_restful_post(JSONObject body, String url, String tag,
 	}
 	
 	public ServerResponse make_restful_post(JSONObject body, String url, String tag, int timeout, boolean log, BranchLinkData linkData) {
-		try {    	
-			if (body.has("app_id") && body.getString("app_id").equals(PrefHelper.NO_STRING_VALUE)) {
-				return new ServerResponse(tag, NO_API_KEY_STATUS);
-			}
+		try {
+			if (body.has(BRANCH_KEY) && body.getString(BRANCH_KEY).equals(PrefHelper.NO_STRING_VALUE)) {
+				return new ServerResponse(tag, NO_BRANCH_KEY_STATUS);
+			} else if (body.has("app_id") && body.getString("app_id").equals(PrefHelper.NO_STRING_VALUE)) {
```

Comment:
Yes. the check for app_id was added back for temporary backward compatibility and meant to be removed in the near future. But, temp code is still code :)